data_generators/job_description_scraper.py

```diff
@@ -564,6 +564,89 @@ def combine_individual_datasets(self, folder_date):
 
         print('Successfully saved combined dataset!')
 
+        return
+
+    def add_job_titles(self, job_list):
+        """
+        Add more job titles to search through.
+
+        Args:
+            job_list (list): A list of job titles in HTML
+            format for the scraper to search through.
+            Remember that empty spaces in words need to
+            be replaced with "%20" to stay HTML compliant.
+                ex: ["web%20developer","backend%20engineer"]
+                ex: ["devops"]
+
+        Returns:
+            None.
+        """
+        # Check if a list was passed
+        if type(job_list) == list:
+            # Add job_list to existing search terms
+            self.search_terms += job_list
+            print("Success.")
+        else:
+            print("Unsuccessful. Please pass a list.")
+
+        return
+
+    def redefine_job_titles(self, job_list):
+        """
+        This will wipe the old search term list
+        and replace it with the list you provide.
+
+        Args:
+            job_list (list): A list of job titles in HTML
+            format for the scraper to search through.
+            Remember that empty spaces in words need to
+            be replaced with "%20" to stay HTML compliant.
+                ex: ["web%20developer","backend%20engineer"]
+                ex: ["devops"]
+
+        Returns:
+            None.
+        """
+        # Check if a list was passed
+        if type(job_list) == list:
+            # Redefine search_term list
+            self.search_terms = job_list
+            print("Success.")
+        else:
+            print("Unsuccessful. Please pass a list.")
+
+        return
+
+    def remove_job_titles(self, job_list):
+        """
+        Remove job titles from the search list. This may
+        produce an error if you provide a search term
+        that does not already exist in the list.
+
+        Args:
+            job_list (list): A list of job titles in HTML
+            format for the scraper to remove.
+            Remember that empty spaces in words need to
+            be replaced with "%20" to stay HTML compliant.
+                ex: ["web%20developer","backend%20engineer"]
+                ex: ["devops"]
+
+        Returns:
+            None.
+        """
+        # Check if a list was passed
+        if type(job_list) == list:
```

Comment:
Checking a type like this isn't the best way to go. If it's actually required it's better to use the is_instance built in function. Also it's better to compare with the generics and not specific types, ie. Sequence for list-like objects. `isinstance(["some list"], Sequence)`. Moreover this type of check is not helping the code very much, it's mostly constricting what our users can do and not providing much benefit. I recommend trying set operations for finding the items in one sequence that aren't in another - just an idea.

Reply:
Done!